flink-next-gen-connector-influxdb/src/main/java/org/apache/flink/streaming/connectors/influxdb/source/reader/InfluxDBSplitReader.java

```diff
@@ -147,26 +149,51 @@ public void handle(final HttpExchange t) throws IOException {
                     }
                 }
 
-                InfluxDBSplitReader.this.ingestionQueue.put(points); // TODO use offer with timeout
+                boolean result =
+                        CompletableFuture.supplyAsync(
+                                        () -> {
+                                            boolean success = false;
+                                            try {
+                                                success =
+                                                        InfluxDBSplitReader.this.ingestionQueue.put(
+                                                                InfluxDBSplitReader.this
+                                                                        .split
+                                                                        .splitId()
+                                                                        .hashCode(),
+                                                                points);
+                                            } catch (InterruptedException e) {
+                                                success = false;
+                                            }
+                                            return success;
```

Comment:
Could you please put this logic inside of a private method? The code is too much indented and hard to read.